lib/bambora/bank/adapters/payment_profile_response.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Bambora
+  module Bank
+    module Adapters
+      ##
+      # Transforms hash keys from camelCase to snake_case and strips vendor-specific prefixes.
+      class PaymentProfileResponse
+        attr_reader :response
+
+        def initialize(response)
+          @response = response
+        end
+
+        def to_h
+          parsed_query_string.each_with_object({}) do |(key, val), obj|
+            obj[transform(key)] = val
+          end
+        end
+
+        private
+
+        def parsed_query_string
+          Bambora::QueryStringResponse.new(response).to_h
+        end
+
+        def transform(camel_case_word)
+          word = camel_case_word.to_s
+          word.gsub!(/([a-z])([A-Z\d])/, '\1_\2')
```

Comment:
That's some regex 👀. Cool usage of regex groupings.

Reply:
Not gonna lie, I looked at the Rails regex for this to be sure I didn't miss anything.